tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -164,6 +189,33 @@ public function testErrorsWithPresenceRequired()
     }
 
     /**
+     * Test that errors() can work with nested data.
+     *
+     * @return void
+     */
+    public function testErrorsWithNestedFields()
+    {
+        $validator = new Validator;
+        $validator->add('user.username', 'letter', ['rule' => 'alphanumeric']);
+        $validator->add('comments.0.comment', 'letter', ['rule' => 'alphanumeric']);
```

Comment:
So addNested() and addNestedMultiple()?

I can make thise changes but they will be more invasive than the current changes as ValidationSet will need to work with fields and nested validators.